### DISCUSSION TITLE
How to get value of Note field that has AppendOnly="TRUE" in the SchemaXml

### DISCUSSION THREAD

**Discussion by Ofer-Gal** [pnp/powershell#655] [Q&A] [answered]
In few libraries I have a field named "DPCNotes" that was created with AppendOnly and now it shows with a link "View Entries" in the views.
When I use in PowerShell 
```
$items = Get-PnPListItem -List $list -Connection $siteConn
foreach ($item in $items) {
    Write-Host $item["DPC Notes"]
}
```
The values are null. 
I intend to replace all the libraries with "DPCNotes" with AppendOnly="FALSE" but before I do it, I need to collect the values in each note to put it back because when you change the field type, the data in the note is lost.
Anyone has a good idea how to overcome this?
Thank you.

Found out it is in the "AppendOnlyHistory" property but even with REST call like: 
`items?&expand=DPCNotes&$select=ID,DPCNotes/AppendOnlyHistory`
I only get the last entry to the field

**Comment by LeonArmston** [comment, marked as answer]
Hi 

These comments are stored as Versions and the way to retrieve the versions is to access the information in the version property of the list item. I quickly made this script below to show the values on your console. if I was you I would modify it to perhaps write the values to a CSV file and list the version comments before changing the field to AppendOnly="FALSE

```powershell
$ctx = Get-PnPContext
$list = Get-PnPList -Identity 'Issue tracker'

$listItems = Get-PnPListItem -List $list

foreach($listItem in $listItems)
{
    $DPCNotes = $null
    $ctx.Load($listItem.Versions)
    $ctx.ExecuteQuery()

    Write-Host $listItem.FieldValues.Title

    foreach($version in $listItem.Versions)
    {
        $DPCNotes += $version["DPCNotes"] # Assign To Variable
    }

    $DPCNotes
}

```

![image](https://user-images.githubusercontent.com/12968962/117537288-d0cfe300-aff7-11eb-8e70-473f141e7990.png)

**Reply by Ofer-Gal** [reply]
Instead of $version["DPCNotes"] the value was in $version["DPC_x0020_Notes"] but your example was a saver. thank you very much

BTW I found out the versions go from latest to earliest. So to reconstruct correctly, you need reverse order.
